### PR TITLE
Test Coverage

### PR DESCRIPTION
**Pull request by guitsaru**
Current Status:

screen shot 2017-01-24 at 21 32 11

### CODE REVIEW

**Comment by veganstraightedge** [comment]
Thanks for taking this on. I know it's a big gnarly task and that I was a bad kid in not writing specs before now. You're a real trooper, @guitsaru.

**Comment by guitsaru** [comment]
@veganstraightedge not a problem; happy to help.

`SimpleCov` now says 100% covered, but that's not completely accurate.  Next step is to go through and write tests for the controllers that haven't been hit yet.

**Comment by veganstraightedge** [comment]
You're gonna remove `WIP` from the PR title when it's ready for review, right?

**Comment by guitsaru** [comment]
@veganstraightedge yep

**Comment by guitsaru** [comment]
@veganstraightedge I think I've got a while to go yet. It might be better to review what's there now to get it in early.  Also it's going to be a huge PR, so splitting it up would be good for review.

Let me know what you think and I can remove the `WIP` now.

File: config/routes.rb [hunk 1]
```diff
@@ -136,7 +136,7 @@
 
 
   # Misc plumbing infrastructure
-  get "manifest.json",  to: "misc#mainfest_json"
+  get "manifest.json",  to: "misc#manifest_json"
```
Comment (veganstraightedge) [inline config/routes.rb]: Great catch.

**Comment by veganstraightedge** [comment]
@guitsaru Since this PR is gonna end up being so big, how about you break it up into smaller chunks? What you've got so far as the first chunk. Then each subsequent PR of specs as it makes sense too.

**Comment by guitsaru** [comment]
@veganstraightedge sounds good.

File: app/helpers/admin_helper.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module AdminHelper
-  def thing_type(thing)
-    request.path.split('admin/').last.split("/").first.capitalize.singularize
+  def thing_type(_thing)
```
Comment (bookis) [inline app/helpers/admin_helper.rb]: Any reason you didn't just get rid of `_thing`? And maybe rename to `current_resource_name` or something.

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -67,6 +67,6 @@ def br_to_p(html)
   end
 
   def homepage?
```
Comment (bookis) [inline app/helpers/application_helper.rb]: Looks like this method is never used. It can be removed.

**Comment by guitsaru** [comment]
@bookis: I agree about the factories, but didn't want to pull too much in without talking about it first and didn't get around to it because of how busy the weekend was.  I'll pull that in in the next phase: it'll be especially useful for the controller testing.

File: app/helpers/admin_helper.rb (hunk 2 above)
Comment (guitsaru) [inline app/helpers/admin_helper.rb]: At the time I was focusing more on testing than changing code, but it's a good idea to take it out.

File: app/helpers/application_helper.rb (hunk 3 above)
Comment (guitsaru) [inline app/helpers/application_helper.rb]: Good call

**Comment by veganstraightedge** [comment]
@guitsaru If this is a good save point, I'll acceptance test then merge PR. Are you good?

**Comment by guitsaru** [comment]
@veganstraightedge go for it

**Pull request merged**